packages/amplication-client/src/Application/code-view/CodeViewExplorerTree.tsx

```diff
@@ -0,0 +1,120 @@
+import { Icon, TreeView } from "@amplication/design-system";
+import React, { useCallback, useEffect, useState } from "react";
+import { Build } from "../../models";
+import { FileDetails } from "./CodeViewPage";
+import { FileExplorerNode } from "./FileExplorerNode";
+import { NodeTypeEnum } from "./NodeTypeEnum";
+import { StorageBaseAxios, StorageResponseType } from "./StorageBaseAxios";
+import { useQuery } from "react-query";
+import { AxiosError } from "axios";
+import { remove } from "lodash";
+import { FileMeta } from "./CodeViewExplorer";
+import "./CodeViewBar.scss";
+
+const CLASS_NAME = "code-view-bar";
+
+type Props = {
+  selectedBuild: Build;
+  onFileSelected: (selectedFile: FileDetails) => void;
+};
+
+const NO_FILES_MESSAGE = "There are no available files to show for this build";
+
+const INITIAL_ROOT_NODE: FileMeta = {
+  type: NodeTypeEnum.Folder,
+  name: "root",
+  path: "/",
+  children: [],
+  expanded: false,
+};
+
+const CodeViewExplorerTree = ({ selectedBuild, onFileSelected }: Props) => {
+  const [rootFile, setRootFile] = useState<FileMeta>(INITIAL_ROOT_NODE);
+  const [selectedFolder, setSelectedFolder] = useState<FileMeta>(rootFile);
+  const [expandedFolders, setExpandedFolders] = useState<string[]>([]);
+
+  const { error, isError } = useQuery<StorageResponseType, AxiosError>(
+    ["storage-folderList", selectedBuild.id, selectedFolder?.path],
+    async () => {
+      return await StorageBaseAxios.instance.folderList(
+        selectedBuild.appId,
+        selectedBuild.id,
+        selectedFolder?.path
+      );
+    },
+    {
+      onSuccess: (data) => {
+        selectedFolder.children = data.result;
+        setRootFile({ ...rootFile });
+      },
+    }
+  );
+
+  useEffect(() => {
+    //reset the state when the build changes
+    const newRootFile = { ...INITIAL_ROOT_NODE };
+    setRootFile(newRootFile);
+    setSelectedFolder(newRootFile);
+  }, [selectedBuild]);
+
+  const handleNodeClick = useCallback(
+    async (file: FileMeta) => {
+      switch (file.type) {
```

Comment:
I would convert switch case to hashMap object. it is efficient in many ways: 
{
  [NodeTypeEnum.File]: () => {
onFileSelected({
            buildId: selectedBuild.id,
            filePath: file.path,
            isFile: true,
            fileName: file.name,
          });
}
}